cmake/doctest.cmake

```diff
@@ -0,0 +1,30 @@
+find_package(doctest 2.4.8 QUIET)
+
+if (doctest_FOUND)
+    message(STATUS "Found doctest: ${doctest_DIR}")
+else(doctest_FOUND)
+    message(STATUS "Failed to find doctest, going to compile from source.")
+    set(DOCTEST_WITH_TESTS OFF CACHE BOOL "doctest tests and examples")
+    set(DOCTEST_WITH_MAIN_IN_STATIC_LIB ON CACHE BOOL "static lib (cmake target) with a default main entry point")
+    #set(DOCTEST_NO_INSTALL OFF CACHE BOOL "Skip the installation process")
+    #set(DOCTEST_USE_STD_HEADERS OFF CACHE BOOL "Use std headers")
+    FetchContent_Declare(
+            doctest
+            GIT_REPOSITORY https://github.com/doctest/doctest
+            GIT_TAG v2.4.11
+            GIT_SHALLOW TRUE # get only the last commit version
+            GIT_PROGRESS TRUE # show progress of download
+            FIND_PACKAGE_ARGS NAMES doctest
```

Comment:
I don't think the `FIND_PACKAGE_ARGS` argument makes sense here, as you manually do a `find_package` call above

Reply:
Looks like the handling of this argument has changed across cmake versions (cmake-3.22 and cmake-3.26 fail in exclusive way: I can fix for one cmake, but then another breaks).
I've comment it out for now. Maybe we'll come back when building recursively.